### ISSUE TITLE
dotnet --info reports duplicates of paths

### ISSUE THREAD

**Issue by ericstj** [dotnet/runtime#59833]
### Description

Run `dotnet --info`

Expect: one instance of each path
Actual: duplicate paths for every entry

```
.NET SDKs installed:
  6.0.100-rc.2.21478.25 [C:\Program Files\dotnet\sdk]
  6.0.100-rc.2.21478.25 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0-rc.2.21475.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-rc.2.21475.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0-rc.2.21474.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-rc.2.21474.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.0-rc.2.21475.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0-rc.2.21475.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Configuration

* Which version of .NET is the code running on? 6.0 RC2
* What OS and version, and what distro if applicable? Windows
* What is the architecture (x64, x86, ARM, ARM64)? All
* Do you know whether it is specific to that configuration? Yes, specific to Windows, not specific to architecture

### Regression?
Yes, previous RC1. 

This is caused by this: https://github.com/dotnet/installer/pull/11843
Previously the bundle would set these properties without trailing slash: https://github.com/dotnet/installer/pull/11843/files#diff-42255eeced0cb7937c40024646092ce85055a361283ed45dec27adbab7996905L124
It was hardcoding the values here, which we had to remove because it couldn't get it right, nor could we calculate it.

This might be different in GA, since we changed how those keys are written again with https://github.com/dotnet/runtime/commit/a7e9f9fb2363e21571da2950990106e7cc4f714e

Need to test GA to see if we should try to change this in the installer or deal with it in the host.

**Comment by ericstj** [comment]
The same repros in GA. Looks like MSI appends that to all directory properties and I don't see a good way to trim it.

One possible fix is to normalize these paths by stripping the slash. Looks like that is already done for the `dotnet_dir` seems reasonable to apply that same normalization to the `global_dirs`
 https://github.com/dotnet/runtime/blob/db1b3023b791aa93d9095699117efb4c0afc41c7/src/native/corehost/hostmisc/utils.cpp#L294-L313

**Comment by ericstj** [comment]
This is going to be a rather visible bug on Windows.  Everyone will hit it (not just ARM64).  I recommend we take it for for 6.0.  The fix should be very low risk.

**Comment by agocke** [comment]
@mateoatr Could you take a look?

**Comment by vitek-karas** [comment]
The change probably didn't make it into dotnet/installer - the SDK -> installer integration has been stuck for a few days: https://github.com/dotnet/installer/pull/12361

**Comment by ericstj** [comment]
Note that the fix for this was approved for both 3.1 and 5.0, however it was after the builds for the November release were complete.

https://github.com/dotnet/runtime/pull/60496
https://github.com/dotnet/core-setup/pull/9186
